src/ucs/sys/sys.c

```diff
@@ -767,23 +767,52 @@ ucs_status_t ucs_sys_get_proc_cap(uint32_t *effective)
 #endif
 }
 
-static void ucs_sysv_shmget_error_check_EPERM(int flags, char *buf, size_t max)
+/**
+ * Checks whether the process have memory lock capability.
+ *
+ * @param [out] mlock_cap Pointer to a flag that indicates whether the
+ *              process have the required capability.
+ *
+ * @return UCS_OK or error in case of failure.
+ */
+
+static ucs_status_t ucs_sys_get_mlock_cap(int *mlock_cap)
 {
 #if HAVE_SYS_CAPABILITY_H
-    ucs_status_t status;
     uint32_t ecap;
+    ucs_status_t status;
 
     UCS_STATIC_ASSERT(CAP_IPC_LOCK < 32);
     status = ucs_sys_get_proc_cap(&ecap);
-    if ((status == UCS_OK) && !(ecap & UCS_BIT(CAP_IPC_LOCK))) {
-        /* detected missing CAP_IPC_LOCK */
-        snprintf(buf, max, ", CAP_IPC_LOCK privilege is needed for SHM_HUGETLB");
-        return;
+
+    if ((status == UCS_OK) && (ecap & UCS_BIT(CAP_IPC_LOCK))) {
+        *mlock_cap = 1;
+    } else {
+        *mlock_cap = 0;
     }
+
+    return status;
+#else
+    return UCS_ERR_UNSUPPORTED;
 #endif
+}
+
+static void ucs_sysv_shmget_error_check_EPERM(int flags, char *buf, size_t max)
+{
+    int mlock_cap = 0;
 
-    snprintf(buf, max,
-             ", please check for CAP_IPC_LOCK privilege for using SHM_HUGETLB");
+    if (ucs_sys_get_mlock_cap(&mlock_cap) != UCS_OK) {
+        snprintf(buf, max,
+                 ", please check for CAP_IPC_LOCK privilege for using "
+                 "SHM_HUGETLB");
+        return;
+    }
+
+    if (!mlock_cap) {
+        /* detected missing CAP_IPC_LOCK */
+        snprintf(buf, max,
+                 ", CAP_IPC_LOCK privilege is needed for SHM_HUGETLB");
```

Comment:
these error messages seem the same

Reply:
you mean we should combine them to a single message?

Reply:
I think there's indeed a difference:
the first message tells the user to check the privilege himself because we didn't succeed.
the second message tells him that the required privilege is missing.
@yosefe

Reply:
i think "please check for CAP_IPC_LOCK privilege for using SHM_HUGETLB" message can cover both cases